src/main/kotlin/nl/avisi/kotlinwebtest/Afterwards.kt

```diff
@@ -0,0 +1,42 @@
+/**
+ * For licensing, see LICENSE.txt
+ * @author Martijn Heuvelink
+ */
+package nl.avisi.kotlinwebtest
+
+interface After<in RequestType : StepRequest, in ResponseType : StepResponse> {
+    fun afterwards(executionContext: ExecutionContext, request: RequestType, response: ResponseType): AfterResult
+
+    fun success(): AfterResult =
+            SuccessAfterResult(this)
+
+    fun failure(message: String): AfterResult =
+            FailureAfterResult(this, message)
+}
+
+interface AfterResult : Result {
+    val validator: After<*, *>
+}
+
+class SuccessAfterResult(override val validator: After<*, *>) : AfterResult {
+    override val success: Boolean = true
```

Comment:
sort of illegalstate, een SuccessAfterResult kan inprincipe ook een success false hebben

Reply:
Hiervoor heb ik FailureAfterResult

Reply:
Laten we het er morgen over hebben :)